tests/test_integration.py

```diff
@@ -2,22 +2,15 @@
 import os
 import unittest
 from pathlib import Path
-from tempfile import TemporaryDirectory, TemporaryFile
+from tempfile import TemporaryDirectory
 
```

Comment:
My goal with this PR is just to extract the `filesystems` submodule from `py-dcqc`. I agree that the test suite can probably be broken up. 

I decided to combine everything into one file as it was in `py-dcqc` (originally named as `test_filesystems.py`). I shouldn't have split it up since that deviates from this PR's original goal, which is just to split off the SynapseFS functionality (ultimately to speed up the `py-dcqc` test suite).